crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -1974,11 +1974,21 @@ async fn payment_response_update_tracker<F: Clone, T: types::Capturable>(
         if payment_intent.status.is_in_terminal_state()
             && business_profile.dynamic_routing_algorithm.is_some()
         {
+            let dynamic_routing_algo_ref: api_models::routing::DynamicRoutingAlgorithmRef =
+                business_profile
+                    .dynamic_routing_algorithm
+                    .clone()
+                    .map(|val| val.parse_value("DynamicRoutingAlgorithmRef"))
+                    .transpose()
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("unable to deserialize DynamicRoutingAlgorithmRef from JSON")?
+                    .unwrap_or_default();
+
             let state = state.clone();
             let business_profile = business_profile.clone();
```

Comment:
you can only clone the `profile_id` instead of whole profile object since below methods only require id
```suggestion
            let profile_id = business_profile.get_id().clone();
```